docs/Module - CI.md

```diff
@@ -1,26 +1,112 @@
-# Module CI
-
-Use the following GitHub workflow to make use of the `module-ci.yaml` reusable workflow. This workflow should be use in every module repository.
-
-Filename: `./github/workflows/module-ci.yaml`
-
-```yaml
-name: module-ci
-on:
-  push:
-    branches:
-      - main
-  pull_request:
-    types:
-      - opened
-      - edited
-      - synchronize
-      - labeled
-      - unlabeled
-jobs:
-  module-ci:
-    uses: cloudeteer/terraform-governance/.github/workflows/module-ci.yaml@main
-    permissions:
-      contents: write
-      pull-requests: read
-```
+# Module CI - Workflow Overview
+
+## Module CI 
+ File : [`.github/workflows/module-ci.yaml`](https://github.com/cloudeteer/terraform-governance/blob/main/.github/workflows/module-ci.yaml)
+
+### Purpose 
+The `module-ci` workflow orchestrates multiple modular workflows for validation, linting, documentation, code analysis, and testing. Additionally, it automates the creation of GitHub issues for failed scheduled workflows.
+
+### Triggers 
+This workflow is triggered using `workflow_call` to allow reuse by other workflows.  
+
+It requires the following secrets:
+- `ARM_CLIENT_ID`: Azure client ID
+- `ARM_SUBSCRIPTION_ID`: Azure subscription ID
+- `ARM_TENANT_ID`: Azure tenant ID
```

Comment:
```suggestion
It requires the following secrets, which are needed for the remote test, which deploys Azure resources to a specific Azure subscription. Instead of using a client secret, an Azure Managed Identity with workflow federation is utilized for deployment:

- `ARM_CLIENT_ID`: Azure Client ID
- `ARM_SUBSCRIPTION_ID`: Azure Subscription ID
- `ARM_TENANT_ID`: Azure Tenant ID